py-polars/tests/unit/test_selectors.py

```diff
@@ -809,3 +809,16 @@ def test_selector_result_order(df: pl.DataFrame, selector: SelectorType) -> None
             "qqR": pl.String,
         }
     )
+
+
+def test_selector_list_of_lists_18499() -> None:
+    lf = pl.DataFrame(
+        {
+            "foo": [1, 2, 3, 1],
+            "bar": ["a", "a", "a", "a"],
+            "ham": ["b", "b", "b", "b"],
+        }
+    )
+
+    with pytest.raises(InvalidOperationError, match="invalid selector expression"):
+        lf.unique(subset=[["bar", "ham"]])
```

Comment:
What kind of expression does this lead to in crates/polars-plan/src/plans/conversion/expr_expansion.rs:L928?

Reply:
The input is parsed as a list with one element: a String Series with two elements (`"bar"`, `"ham"`). Which is correct.

So it becomes a `Literal`, not a `Column`.